client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -206,63 +233,92 @@ tie.factory('PythonCodePreprocessorService', [
       var fullTestCode = [
         'def matches_buggy_function(func):',
         '    buggy_results = []',
-        '    for test_input in test_inputs:',
-        '        buggy_results.append(' + testOutputCode + ')',
+        '    for task_tests in all_tasks_test_inputs:',
+        '        task_results = []',
+        '        for test_input in task_tests:',
+        '            task_results' +
+        '.append(' + testOutputCode + ')',
+        '        buggy_results.append(task_results)',
         '    return buggy_results == ' + VARNAME_CORRECTNESS_TEST_RESULTS,
         '',
         VARNAME_BUGGY_OUTPUT_TEST_RESULTS + ' = []',
         ''
       ].join('\n');
-      buggyOutputTests.forEach(function(buggyOutputTest) {
-        var qualifiedBuggyFunctionName = (
-          buggyOutputTest.getBuggyFunctionName());
+
+      buggyOutputTests.forEach(function(oneTaskBuggyOutputTest) {
```

Comment:
Perhaps this ought to be: `allTasksBuggyOutputTests.forEach(function(oneTaskBuggyOutputTests) {`. Note the pluralization of Tests at the end.